src/oc_erchef/apps/oc_chef_wm/src/chef_wm_actions_queue_monitoring.erl

```diff
@@ -0,0 +1,473 @@
+    try
+        gen_server:call(?SERVER, is_queue_at_capacity, Timeout)
+    catch Error:Reason ->
+        lager:warning("Queue monitor timeout ~p ~p ", [Error, Reason]),
+        true
+    end.
+
+% log that a message has been dropped (hasn't been sent to rabbitmq)
+% This is a cast so we don't wait for an `ok` response
+% back to the callee, which a call would have returned as well.
+message_dropped() ->
+    gen_server:cast(?SERVER, message_dropped).
+
+
+-spec stop() -> ok.
+stop() ->
+    gen_server:call(?SERVER, stop).
+
+
+%% Support functions --------------------------------------------
+% Synchronously update state by pinging the RabbitMQ Management Console
+sync_check_current_state() ->
+    gen_server:call(?SERVER, sync_check_current_state).
+
+% manually toggle the at capacity flag
+override_queue_at_capacity(AtCapacity) ->
+    gen_server:call(?SERVER, {override_queue_at_capacity, AtCapacity}).
+
+start_timer() ->
+    gen_server:call(?SERVER, start_timer).
+
+stop_timer() ->
+    gen_server:call(?SERVER, stop_timer).
+
+
+%% oc_httpc pool functions --------------------------------------------
+create_pool() ->
+    Pools = get_pool_configs(),
+    [oc_httpc:add_pool(PoolNameAtom, Config) || {PoolNameAtom, Config} <- Pools, Config /= []],
+    ok.
+
+delete_pool() ->
+    Pools = get_pool_configs(),
+    [ok = oc_httpc:delete_pool(PoolNameAtom) || {PoolNameAtom, _Config} <- Pools],
+    ok.
+
+get_pool_configs() ->
+    Config = envy:get(oc_chef_wm, ?POOLNAME, [], any),
+    [{?POOLNAME, Config}].
+
+%%-------------------------------------------------------------
+
+init([]) ->
+    % used to catch worker msgs
+    process_flag(trap_exit, true),
+    TRef = start_update_timer(),
+    {ok, #queue_monitor_state{timer=TRef, worker_process = undefined}}.
+
+
+%%-------------------------------------------------------------
+%% CALLS
+handle_call(is_queue_at_capacity, _From, #queue_monitor_state{queue_at_capacity =
+                                                QueueAtCapacity} = State) ->
+    {reply, QueueAtCapacity, State};
+
+handle_call({override_queue_at_capacity, AtCapacity}, _From, State) ->
+    lager:info("Manually setting Queue Monitor queue_at_capacity ~p", [AtCapacity]),
+    {reply, ok, State#queue_monitor_state{queue_at_capacity = AtCapacity}};
+handle_call(status, _From, State) ->
+    % return a term to be converted to JSON
+    Stats = [{queue_at_capacity,State#queue_monitor_state.queue_at_capacity},
+             {dropped_since_last_check, State#queue_monitor_state.dropped_since_last_check},
+             {max_length, State#queue_monitor_state.max_length},
+             {last_recorded_length, State#queue_monitor_state.last_recorded_length},
+             {total_dropped, State#queue_monitor_state.total_dropped}],
+    {reply, Stats, State};
+handle_call(sync_check_current_state, From, #queue_monitor_state{sync_response_process = undefined} = State) ->
```

Comment:
If a status_ping is already in flight, this will cause a second one to be created. In that case could we update the state to capture it and not issue the message, so that the in-flight request will service the `sync_check_current_state` request?